system/EntityCast/CastAsBoolean.php

```diff
@@ -22,6 +22,6 @@ class CastAsBoolean extends AbstractCast
 	 */
 	public static function get($value, array $params = []): bool
 	{
-		return (bool) $value;
+		return filter_var($value, FILTER_VALIDATE_BOOLEAN);
```

Comment:
Probably use `is_string($value) && in_array($value, ['true', 'false', 'yes', 'no'], true)` check first for the filter_var(), and fallback to `(bool) $value` otherwise ?

```suggestion
              if (is_string($value) && in_array($value, ['true', 'false', 'yes', 'no'], true))
              {
                  return filter_var($value, FILTER_VALIDATE_BOOLEAN);
              }

             return (bool) $value;
```

Reply:
but it will also require strtolower($value) to handle it case insensitive

I suppose this is due to handling integers as true

What about doing 

```
filter_var($value, FILTER_VALIDATE_BOOLEAN, FILTER_NULL_ON_FAILURE) ?? true;
```

From PHP doc

If FILTER_NULL_ON_FAILURE is set, false is returned only for "0", "false", "off", "no", and "", and null is returned for all non-boolean values.

Reply:
I just did test with FILTER_NULL_ON_FAILURE it will return false only for the following

```
null
0
'0'
'no' (case insensitive)
'false' (case insensitive)
```

Reply:
The only difference I see with this approach is 

```
$value = [];
var_dump((bool)$value); //false
var_dump(filter_var($value, FILTER_VALIDATE_BOOLEAN, FILTER_NULL_ON_FAILURE) ?? true); //true
````

I'm not sure which one should be correct behavior but can be easily solved with:
```
var_dump(filter_var($value ?: false, FILTER_VALIDATE_BOOLEAN, FILTER_NULL_ON_FAILURE) ?? true);
```